app/src/main/java/ru/ifmo/md/extratask1/ResultsList.java

```diff
@@ -0,0 +1,178 @@
+package ru.ifmo.md.extratask1;
+
+import android.content.res.Configuration;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.ActionBarActivity;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ProgressBar;
+
+import java.util.List;
+
+public class ResultsList extends ActionBarActivity {
+    static int IMAGE_LOADING_TIMEOUT = 60 * 1000;
+
+    DbHelper helper;
+    ViewPager pager;
+    MyFragmentPagerAdapter adapter;
+    List<Image> allImages;
+    int imagesLoaded;
+
+    MenuItem updateButton;
+    ProgressBar progressBar;
+
+    int curPage;
+    int pageCount;
+    int picsPerPage;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_results_list);
+
+        int prevPage = 0;
+        int prevPicsPerPage = 0;
+        if (savedInstanceState != null) {
+            prevPage = savedInstanceState.getInt("curPage");
+            prevPicsPerPage = savedInstanceState.getInt("prevPPP");
+        }
+
+        pager = (ViewPager) findViewById(R.id.pager);
+        adapter = new MyFragmentPagerAdapter(getSupportFragmentManager());
+
+        pager.setOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                setCurPage(position);
+            }
+
+            @Override
+            public void onPageScrollStateChanged(int state) {
+            }
+        });
+
+        if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
+            picsPerPage = 10;
+            pageCount = 6;
+        } else {
+            picsPerPage = 6;
+            pageCount = 10;
+        }
+        setCurPage((prevPage + 1) * prevPicsPerPage / picsPerPage);
+
+        progressBar = (ProgressBar) findViewById(R.id.progress);
+
+        helper = new DbHelper(this);
+
+        List<Image> images = helper.getImages();
+        if (images.size() == 0) {
+            updatePhotos();
+        } else {
+            onImageSearchFinished(images);
+        }
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle savedInstanceState) {
+        savedInstanceState.putInt("curPage", curPage);
```

Comment:
super.onSaveInstanceState() is not called chiefly to prevent ViewPager to remember its current page, so we could later set it with setCurrentItem (line 151). Is there more decent way to do the same?